hamlet-cli/hamlet/command/schema/__init__.py

```diff
@@ -0,0 +1,224 @@
+
+def find_schemas_from_options(generation, deployment_group, deployment_units):
+    query_args = {
+        'deployment_mode': '',
+        'generation_entrance': 'schemaset',
+        'generation_input_source': generation.generation_input_source,
+        'generation_provider': generation.generation_provider,
+        'generation_framework': generation.generation_framework,
+        'output_filename': 'schemaset-' + deployment_group + '-schemacontract.json',
+        'refresh_output': True
+    }
+    try:
+        available_schemas = query_backend.run(
+            **query_args,
+            cwd=os.getcwd(),
+            query_text=LIST_SCHEMAS_QUERY
+        )
+
+    except BackendException as e:
+        raise CommandError(str(e))
+
+    deployments = []
+
+    for schema in available_schemas:
+        if re.fullmatch(deployment_group, schema['DeploymentGroup']):
+            for deployment_unit in deployment_units:
+                if re.fullmatch(deployment_unit, schema['DeploymentUnit']):
+                    deployments.append(schema)
+
+    return deployments
+
+class Generation(object):
+    def __init__(self, generation_provider=None, generation_framework=None, generation_input_source=None):
+        self.generation_provider = generation_provider
+        self.generation_framework = generation_framework
+        self.generation_input_source = generation_input_source
+
+
+pass_generation = click.make_pass_decorator(Generation, ensure=True)
+
+@cli.group('schema')
+@click.pass_context
+@click.option(
+    '-p',
+    '--generation-provider',
+    help='provider for output generation',
+    default=['aws'],
+    multiple=True,
+    show_default=True
+)
+@click.option(
+    '-f',
+    '--generation-framework',
+    help='output framework to use for output generation',
+    default='cf',
+    show_default=True
+)
+@click.option(
+    '-i',
+    '--generation-input-source',
+    help='source of input data to use when generating the output',
+    default='composite',
+    show_default=True
+)
+def group(ctx, generation_provider, generation_framework, generation_input_source):
+    """
+    Generates JSONSchema files for Hamlet data types
+    """
+    ctx.obj = Generation(
+        generation_provider=generation_provider,
+        generation_framework=generation_framework,
+        generation_input_source=generation_input_source
+    )
+
+LIST_SCHEMAS_QUERY = (
+    'Stages[].Steps[]'
+    '.{'
+    'DeploymentGroup:Parameters.DeploymentGroup,'
+    'DeploymentUnit:Parameters.DeploymentUnit,'
+    'DeploymentProvider:Parameters.DeploymentProvider'
+    '}'
+)
+
+def schema_table(data):
+    tablerows = []
+    for row in data:
+        tablerows.append(
+            [
+                wrap_text(row['DeploymentGroup']),
+                wrap_text(row['DeploymentUnit']),
+            ]
+        )
+    return tabulate(
+        tablerows,
+        headers=['DeploymentGroup', 'DeploymentUnit'],
+        tablefmt='github'
+    )
+
```

Comment:
Should these be renamed to reflect what they actually are? 

DeploymentGroup = Type 
DeploymentUnit = Instance 

or something like that

Reply:
You are choosing the output values in the file being queried at this point. So I think it would be better to actually update the output to reflect actual names rather than align with the way input is being passed in

Reply:
Yep, re-aligned with their original intent. Cheers.